cpp/inc/bond/protocol/simple_json_writer.h

```diff
@@ -195,10 +195,10 @@ class SimpleJsonWriter
     void WriteUnicode(wchar_t c)
     {
         char u[6] = "\\u";
-        u[2] = detail::HexDigit(c >> 12);
-        u[3] = detail::HexDigit(c >> 8);
-        u[4] = detail::HexDigit(c >> 4);
-        u[5] = detail::HexDigit(c >> 0);
+        u[2] = detail::HexDigit((char)(c >> 12));
```

Comment:
It looks like we actually want to call the `HexDigit(int)` overload instead of the `HexDigit(char)` one.
This function is turning a 32-bit `wchar_t` into its JSON-escaped `"\u0A1B"` form.